### PR TITLE
[expo-go] Remove vendored view shot

### PR DESCRIPTION
**Pull request by alanjhughes**
# Why
View shot was still vendored in expo go. 

# How
- Removed the vendored code
- Added view shot to `expoGoConfig` and created a patch.

`et umv -m react-native-view-shot -c v4.0.2`

I had to change the view shot constructor because in the vendored code we accepted a `ScopedContext` parameter but doing this now would require the module to take a dependency on `expoview` which would cause a circular dependency. Ideally, shared code should be taken out of `expoview` and moved to a separate module but this is a significant refactor. 

# Test Plan
Expo go. ViewShot is working correctly on both iOS and Android

### CODE REVIEW

File: tools/src/vendoring/config/expoGoConfig.ts [hunk 1]
```diff
@@ -13,7 +13,30 @@ const config: VendoringTargetConfig = {
       source: 'https://github.com/aws-amplify/amplify-js.git',
     },
     'react-native-view-shot': {
-      source: 'https://github.com/gre/react-native-view-shot.git',
+      source: 'react-native-view-shot',
+      sourceType: 'npm',
+      excludeFiles: ['src/__tests__/**/*', 'windows/**/*', 'LICENSE', 'README.md'],
```
Comment (Kudo) [inline tools/src/vendoring/config/expoGoConfig.ts:18]:
```suggestion
      excludeFiles: ['src/__tests__/**/*', 'windows/**/*', 'LICENSE', 'README.md'],
```

i had README and LICENSE in webview and async-storage. we can align this

**Review by Kudo — APPROVED** [review]

File: tools/src/vendoring/config/expoGoConfig.ts (hunk 1 above)
Comment (alanjhughes) [reply tools/src/vendoring/config/expoGoConfig.ts:18]: Leave them there or remove them from the other modules?

Comment (Kudo) [reply tools/src/vendoring/config/expoGoConfig.ts:18]: my original thought is just about to leave the local modules/ as similar as the package downloaded in node_modules/. not a strong opinion and open to get more feedback. which is your preference btw?

Comment (alanjhughes) [reply tools/src/vendoring/config/expoGoConfig.ts:18]: I was just going to remove unnecessary files but I don't have a strong opinion on it either. We can leave them there. We won't be vendoring that many modules like this

Comment (Kudo) [reply tools/src/vendoring/config/expoGoConfig.ts:18]: alright! let the files back. if that files change too much and pollute our git history. we can remove them later.

**Review by Kudo — APPROVED** [review]
thanks again!

**Review by lukmccall — APPROVED** [review]

**Pull request merged**